workspace.spec

```diff
@@ -572,6 +572,359 @@ module Workspace {
 		string description;
 	} ProvenanceAction;
 
+	/* Credit Metadata-related objects and fields
+
+		Workspace objects have a list of zero or more sets of credit metadata (CM), or citation information for the object. This information allows data imported into KBase to retain the appropriate details for users wishing to cite the data source, and for those who contributed to its creation to be credited for their work.
+
+		CM is stored in a list as part of the object provenance, but unlike other provenance information, which is immutable, CM can be updated.
+
+		Credit metadata is added after object creation, and requires the user to be a workspace admin. Username, timestamp, and credit metadata schema version are stored along with the CM, which allows tracking of CM changes.
+
+		To update existing CM, a new set of credit metadata is added to the existing list.
+
+		When requesting credit metadata, the last (i.e. most recent) entry in the list is returned.
+
+		In the following entries, "resource" is used to refer to the workspace object that the CM pertains to.
+	*/
```

Comment:
this blurb is not written in a particularly fluid manner -- any suggestions to make it sound a bit less lumpen gladly received

Reply:
updates - "new set" is a little confusing. Implies the prev record is archived and a new record is used. But maybe that is correct, given the line after. So if someone added an author, we wouldn't just append the record w/ a change log, we would archive the prev list, and provide a new copy of the list with the addition.

Reply:
> Implies the prev record is archived and a new record is used. But maybe that is correct, given the line after.

That is correct -- it's just that we're saving all the records in a list, with the last entry in the list being the most recent. To update credit metadata, you would upload a new set of credit metadata, which would get added to the end of the list. The last list item is returned when there is a request for credit metadata. If there was a need to track down when some change was made to the metadata, we can look through the list entries to see when it first appeared.

Reply:
does that require some to have the full credit metadata record to resubmit?

Reply:
yep. I've updated the explanation to clarify.